presto-main/src/main/java/com/facebook/presto/cost/CostCalculatorUsingExchanges.java

```diff
@@ -238,6 +239,12 @@ public PlanNodeCostEstimate visitLimit(LimitNode node, Void context)
             return cpuCost(getStats(node).getOutputSizeInBytes(node.getOutputSymbols(), types));
         }
 
+        @Override
+        public PlanNodeCostEstimate visitRemoteSource(RemoteSourceNode node, Void context)
+        {
+            return ZERO_COST;
```

Comment:
Add a comment here explaining why remote sources are ignored in cost estimation.

of course, throwing would be safer here, in case cost calculator is used on actual plan somewhere else than in EXPLAIN.

Reply:
I was just making this a pass through so we could get the costs for other things, but now I'm thinking that we'll lose the exchange cost. Maybe I should make a FragmentCostCalculator extension to get the exchange properties from the fragment and then call `calculateExchangeCost`

Reply:
What about different approach -- that we de-fragment the plan and pass that to the cost calculator? Then the stats & costing code would not need to be modified.

Reply:
> I was just making this a pass through so we could get the costs for other things, but now I'm thinking that we'll lose the exchange cost.

I didn't notice. The problem is that it's possible for "cost on fragmented plan" to be very off, and it may go unnoticed for long time: are we able to test-cover everything? will we notice in production when costs are off? (ultimately we will, but not too soon, as the costing here won't have any impact on execution)

Reply:
I had looked at defragmenting and computing the cost, but it gets very messy in the plan printer because each node you visit needs to have the defragmented plan and match up correctly (or re.-defragment), etc. 

I'm going to go with the FragmentedPlanCostCalculator approach for handling RemoteSourceNodes and I'll add a test that creates a plan and computes the cost and then fragments the plan and checks that the cumulative cost is the same.